# Imports dependencies

In [2]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.9.1


# Load dataset

In [3]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [4]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    def ticket_number(x):
        return x.split(" ")[-1]
        
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)                     
    return df
    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                              Name     Sex   Age  SibSp  \
0                            Braund Mr Owen Harris    male  22.0      1   
1  Cumings Mrs John Bradley Florence Briggs Thayer  female  38.0      1   
2                             Heikkinen Miss Laina  female  26.0      0   
3         Futrelle Mrs Jacques Heath Lily May Peel  female  35.0      1   
4                           Allen Mr William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked Ticket_number Ticket_item  
0      0         A/5 21171   7.2500   NaN        S         21171         A/5  
1      0          PC 17599  71.2833   C85        C         17599          PC  
2      0  STON/O2. 3101282   7.9250   NaN        S       3101282    STON/O2.  
3      0            113803  53.1000  C123        S        113803        NONE  
4      0            373450   8.0500   NaN        S        373450        NONE

Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [5]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived")
#input_features.remove("Ticket_number")

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin', 'Embarked', 'Ticket_number', 'Ticket_item']


# Convert Pandas dataset to TensorFlow Dataset

In [6]:
def tokenize_names(features, labels=None):
    """Divite the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[WARNING 24-12-16 08:25:59.1421 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:25:59.1432 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:25:59.1432 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:26:03.9869 UTC kernel.cc:1233] Loading model from path /tmp/tmp7rs6e_0v/model/ with prefix 3f42c84947d54446
[INFO 24-12-16 08:26:03.9926 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-16 08:26:03.9931 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:26:03.9931 UTC kernel.cc:1061] Use fast generic engine


Accuracy: 0.8260869383811951 Loss:0.8608942627906799


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [8]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    # compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    # hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    #max_depth=4,
    shrinkage=0.05,
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=2000,
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[WARNING 24-12-16 08:26:09.3349 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:26:09.3349 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:26:09.3349 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:26:10.2065 UTC kernel.cc:1233] Loading model from path /tmp/tmpxuzpbamd/model/ with prefix 0207f3d71c8645cc
[INFO 24-12-16 08:26:10.2143 UTC decision_forest.cc:734] Model loaded with 40 root(s), 2106 node(s), and 10 input feature(s).
[INFO 24-12-16 08:26:10.2144 UTC kernel.cc:1061] Use fast generic engine


Accuracy: 0.782608687877655 Loss:1.0586705207824707


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [9]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1 (1.00 Byte)
Trainable params: 0 (0.00 Byte)
Non-trainable params: 1 (1.00 Byte)
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (11):
	Age
	Cabin
	Embarked
	Fare
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_item
	Ticket_number

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.           "Sex"  0.585997 ################
    2.           "Age"  0.364636 #######
    3.          "Fare"  0.266191 ###
    4.          "Name"  0.207054 #
    5.        "Pclass"  0.179191 
    6. "Ticket_number"  0.178806 
    7.      "Embarked"  0.177803 
    8.   "Ticket_item"  0.177009 
    9.         "Parch"  0.175276 
   10.         "SibSp"  0.171694 

Variable Importance: NUM_AS_ROOT:
    1.  "Sex" 34.000000 ######

# Make predictions

In [10]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,0
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [11]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=0)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmp0hujnfnr as temporary training directory


[WARNING 24-12-16 08:26:11.8412 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:26:11.8412 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:26:11.8413 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:34.9964 UTC kernel.cc:1233] Loading model from path /tmp/tmp0hujnfnr/model/ with prefix a5c2d7bd1c474c81
[INFO 24-12-16 08:31:35.0165 UTC decision_forest.cc:734] Model loaded with 30 root(s), 2236 node(s), and 12 input feature(s).
[INFO 24-12-16 08:31:35.0166 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesGeneric" built
[INFO 24-12-16 08:31:35.0166 UTC kernel.cc:1061] Use fast generic engine


Accuracy: 0.8630136847496033 Loss:0.6749962568283081


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [12]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[WARNING 24-12-16 08:31:35.4036 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:35.4036 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:35.4037 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:35.9043 UTC kernel.cc:1233] Loading model from path /tmp/tmpidyjww8j/model/ with prefix 6297e62f25434877
[INFO 24-12-16 08:31:35.9082 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-16 08:31:35.9084 UTC kernel.cc:1061] Use fast generic engine


i:1


[WARNING 24-12-16 08:31:36.3384 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:36.3384 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:36.3384 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:37.1807 UTC kernel.cc:1233] Loading model from path /tmp/tmpj0qktz26/model/ with prefix e4e65afc0ae54960
[INFO 24-12-16 08:31:37.1989 UTC kernel.cc:1061] Use fast generic engine


i:2


[WARNING 24-12-16 08:31:37.6339 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:37.6339 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:37.6339 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:38.1371 UTC kernel.cc:1233] Loading model from path /tmp/tmpc357i_w7/model/ with prefix 872ff4a9f3064455
[INFO 24-12-16 08:31:38.1414 UTC kernel.cc:1061] Use fast generic engine


i:3


[WARNING 24-12-16 08:31:38.5709 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:38.5709 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:38.5710 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:39.6433 UTC kernel.cc:1233] Loading model from path /tmp/tmpu9jc3yce/model/ with prefix c7063856fd534381
[INFO 24-12-16 08:31:39.6680 UTC kernel.cc:1061] Use fast generic engine


i:4


[WARNING 24-12-16 08:31:40.1007 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:40.1008 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:40.1008 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:40.6490 UTC kernel.cc:1233] Loading model from path /tmp/tmp2ng3hvq2/model/ with prefix 8cdf9d152f9240f3
[INFO 24-12-16 08:31:40.6544 UTC kernel.cc:1061] Use fast generic engine


i:5


[WARNING 24-12-16 08:31:41.0835 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:41.0836 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:41.0836 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:41.5490 UTC kernel.cc:1233] Loading model from path /tmp/tmpqvqoujvs/model/ with prefix 78597ad61bc24d5d
[INFO 24-12-16 08:31:41.5519 UTC kernel.cc:1061] Use fast generic engine


i:6


[WARNING 24-12-16 08:31:41.9928 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:41.9928 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:41.9929 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:42.5784 UTC kernel.cc:1233] Loading model from path /tmp/tmp8igsf__b/model/ with prefix e8844e334ce14d58
[INFO 24-12-16 08:31:42.5856 UTC kernel.cc:1061] Use fast generic engine


i:7


[WARNING 24-12-16 08:31:43.0183 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:43.0184 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:43.0184 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:43.9441 UTC kernel.cc:1233] Loading model from path /tmp/tmps0otbs6n/model/ with prefix 1371e661778247d1
[INFO 24-12-16 08:31:43.9631 UTC kernel.cc:1061] Use fast generic engine


i:8


[WARNING 24-12-16 08:31:44.4060 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:44.4060 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:44.4061 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:45.0537 UTC kernel.cc:1233] Loading model from path /tmp/tmp2pwjbqm6/model/ with prefix 52f4a784ee3540a5
[INFO 24-12-16 08:31:45.0630 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:31:45.0630 UTC kernel.cc:1061] Use fast generic engine


i:9


[WARNING 24-12-16 08:31:45.4935 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:45.4936 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:45.4936 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:46.2520 UTC kernel.cc:1233] Loading model from path /tmp/tmpiq6868vl/model/ with prefix 40abb2dac6164c01
[INFO 24-12-16 08:31:46.2655 UTC kernel.cc:1061] Use fast generic engine


i:10


[WARNING 24-12-16 08:31:46.7089 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:46.7089 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:46.7090 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:47.2560 UTC kernel.cc:1233] Loading model from path /tmp/tmpkrcf1ke5/model/ with prefix 9282b3e1db35462a
[INFO 24-12-16 08:31:47.2614 UTC kernel.cc:1061] Use fast generic engine


i:11


[WARNING 24-12-16 08:31:47.6985 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:47.6986 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:47.6986 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:48.4559 UTC kernel.cc:1233] Loading model from path /tmp/tmpm3quzhl8/model/ with prefix 1067003046294ef1
[INFO 24-12-16 08:31:48.4696 UTC kernel.cc:1061] Use fast generic engine


i:12


[WARNING 24-12-16 08:31:48.9104 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:48.9104 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:48.9104 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:49.4434 UTC kernel.cc:1233] Loading model from path /tmp/tmpwyxl0lbn/model/ with prefix dd301db74e9c4aae
[INFO 24-12-16 08:31:49.4489 UTC kernel.cc:1061] Use fast generic engine


i:13


[WARNING 24-12-16 08:31:49.8835 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:49.8835 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:49.8835 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:50.9894 UTC kernel.cc:1233] Loading model from path /tmp/tmp4d808b83/model/ with prefix 70a909ac5893418a
[INFO 24-12-16 08:31:51.0020 UTC kernel.cc:1061] Use fast generic engine


i:14


[WARNING 24-12-16 08:31:51.4780 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:51.4780 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:51.4780 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:52.0631 UTC kernel.cc:1233] Loading model from path /tmp/tmpa14bvbzl/model/ with prefix 3ed7c845c8784619
[INFO 24-12-16 08:31:52.0696 UTC kernel.cc:1061] Use fast generic engine


i:15


[WARNING 24-12-16 08:31:52.5375 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:52.5375 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:52.5375 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:53.1249 UTC kernel.cc:1233] Loading model from path /tmp/tmpcbnc6ano/model/ with prefix 9e36686bf4cb4788
[INFO 24-12-16 08:31:53.1324 UTC kernel.cc:1061] Use fast generic engine


i:16


[WARNING 24-12-16 08:31:53.5708 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:53.5708 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:53.5708 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:54.2661 UTC kernel.cc:1233] Loading model from path /tmp/tmpmfkgpsrd/model/ with prefix d5bfa17b96af48e8
[INFO 24-12-16 08:31:54.2787 UTC kernel.cc:1061] Use fast generic engine


i:17


[WARNING 24-12-16 08:31:54.7187 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:54.7187 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:54.7187 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:55.4286 UTC kernel.cc:1233] Loading model from path /tmp/tmph5fxw_ep/model/ with prefix e5835ca5eb6d443b
[INFO 24-12-16 08:31:55.4411 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:31:55.4411 UTC kernel.cc:1061] Use fast generic engine


i:18


[WARNING 24-12-16 08:31:55.8735 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:55.8736 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:55.8736 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:56.5390 UTC kernel.cc:1233] Loading model from path /tmp/tmpt0w55ysb/model/ with prefix 60188ab4e5cd4680
[INFO 24-12-16 08:31:56.5499 UTC kernel.cc:1061] Use fast generic engine


i:19


[WARNING 24-12-16 08:31:56.9754 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:56.9754 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:56.9754 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:57.8181 UTC kernel.cc:1233] Loading model from path /tmp/tmper_sni7f/model/ with prefix 66739750dedd4b40
[INFO 24-12-16 08:31:57.8351 UTC kernel.cc:1061] Use fast generic engine


i:20


[WARNING 24-12-16 08:31:58.2658 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:58.2658 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:58.2658 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:59.0268 UTC kernel.cc:1233] Loading model from path /tmp/tmp5qktddq9/model/ with prefix eab12722bbf44564
[INFO 24-12-16 08:31:59.0408 UTC kernel.cc:1061] Use fast generic engine


i:21


[WARNING 24-12-16 08:31:59.4710 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:59.4710 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:31:59.4710 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:31:59.9747 UTC kernel.cc:1233] Loading model from path /tmp/tmpsgcpjws0/model/ with prefix 3830eb7814c34d5e
[INFO 24-12-16 08:31:59.9796 UTC kernel.cc:1061] Use fast generic engine


i:22


[WARNING 24-12-16 08:32:00.4268 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:00.4269 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:00.4269 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:01.0317 UTC kernel.cc:1233] Loading model from path /tmp/tmpsfub59to/model/ with prefix 6ada436746d74862
[INFO 24-12-16 08:32:01.0398 UTC kernel.cc:1061] Use fast generic engine


i:23


[WARNING 24-12-16 08:32:01.4800 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:01.4801 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:01.4801 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:02.0970 UTC kernel.cc:1233] Loading model from path /tmp/tmpu8nv19la/model/ with prefix 1f8e06c8729d43f5
[INFO 24-12-16 08:32:02.1052 UTC kernel.cc:1061] Use fast generic engine


i:24


[WARNING 24-12-16 08:32:02.5415 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:02.5415 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:02.5415 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:03.0873 UTC kernel.cc:1233] Loading model from path /tmp/tmp_k819cdy/model/ with prefix 92b18e8caf2445f3
[INFO 24-12-16 08:32:03.0928 UTC kernel.cc:1061] Use fast generic engine


i:25


[WARNING 24-12-16 08:32:03.5270 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:03.5270 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:03.5270 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:04.2153 UTC kernel.cc:1233] Loading model from path /tmp/tmpr8nk_dzm/model/ with prefix 0517d92461304635
[INFO 24-12-16 08:32:04.2262 UTC kernel.cc:1061] Use fast generic engine


i:26


[WARNING 24-12-16 08:32:04.6627 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:04.6627 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:04.6628 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:05.2995 UTC kernel.cc:1233] Loading model from path /tmp/tmpl0n4ow6l/model/ with prefix bb513ef23b904f97
[INFO 24-12-16 08:32:05.3087 UTC kernel.cc:1061] Use fast generic engine


i:27


[WARNING 24-12-16 08:32:05.7399 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:05.7399 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:05.7399 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:06.2997 UTC kernel.cc:1233] Loading model from path /tmp/tmpsees1qt1/model/ with prefix bf7808065e444df7
[INFO 24-12-16 08:32:06.3055 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-16 08:32:06.3060 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:32:06.3060 UTC kernel.cc:1061] Use fast generic engine


i:28


[WARNING 24-12-16 08:32:06.7432 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:06.7432 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:06.7432 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:07.2756 UTC kernel.cc:1233] Loading model from path /tmp/tmpos20lgfh/model/ with prefix e52f4de3806f4be9
[INFO 24-12-16 08:32:07.2806 UTC kernel.cc:1061] Use fast generic engine


i:29


[WARNING 24-12-16 08:32:07.7206 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:07.7207 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:07.7207 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:08.4610 UTC kernel.cc:1233] Loading model from path /tmp/tmp1y4py70t/model/ with prefix 44bd6ff47f334705
[INFO 24-12-16 08:32:08.4732 UTC kernel.cc:1061] Use fast generic engine


i:30


[WARNING 24-12-16 08:32:08.9126 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:08.9126 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:08.9126 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:10.0030 UTC kernel.cc:1233] Loading model from path /tmp/tmphsmixjwn/model/ with prefix 74570e2abd404469
[INFO 24-12-16 08:32:10.0284 UTC kernel.cc:1061] Use fast generic engine


i:31


[WARNING 24-12-16 08:32:10.4718 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:10.4719 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:10.4719 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:11.1206 UTC kernel.cc:1233] Loading model from path /tmp/tmpna76b93p/model/ with prefix ced0bc02414c49a7
[INFO 24-12-16 08:32:11.1298 UTC kernel.cc:1061] Use fast generic engine


i:32


[WARNING 24-12-16 08:32:11.5619 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:11.5619 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:11.5619 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:12.0914 UTC kernel.cc:1233] Loading model from path /tmp/tmpn__qbtil/model/ with prefix 73d2e8d4752b4460
[INFO 24-12-16 08:32:12.0968 UTC kernel.cc:1061] Use fast generic engine


i:33


[WARNING 24-12-16 08:32:12.5370 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:12.5370 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:12.5371 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:13.2692 UTC kernel.cc:1233] Loading model from path /tmp/tmpm5ckct_h/model/ with prefix 377c79a8de814b24
[INFO 24-12-16 08:32:13.2812 UTC kernel.cc:1061] Use fast generic engine


i:34


[WARNING 24-12-16 08:32:13.7204 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:13.7205 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:13.7205 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:14.3753 UTC kernel.cc:1233] Loading model from path /tmp/tmp8uiur5w8/model/ with prefix 3adf48bfd13a4fc2
[INFO 24-12-16 08:32:14.3835 UTC kernel.cc:1061] Use fast generic engine


i:35


[WARNING 24-12-16 08:32:14.8408 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:14.8408 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:14.8409 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:16.0033 UTC kernel.cc:1233] Loading model from path /tmp/tmpofs4d10r/model/ with prefix 17c4564410994b99
[INFO 24-12-16 08:32:16.0132 UTC kernel.cc:1061] Use fast generic engine


i:36


[WARNING 24-12-16 08:32:16.5316 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:16.5316 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:16.5317 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:17.3255 UTC kernel.cc:1233] Loading model from path /tmp/tmpm58y2uxx/model/ with prefix 9d2657d5eaee4db3
[INFO 24-12-16 08:32:17.3419 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:32:17.3419 UTC kernel.cc:1061] Use fast generic engine


i:37


[WARNING 24-12-16 08:32:17.8270 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:17.8270 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:17.8270 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:18.4630 UTC kernel.cc:1233] Loading model from path /tmp/tmpchrsrn6s/model/ with prefix 5fb14f8c38ba4489
[INFO 24-12-16 08:32:18.4724 UTC kernel.cc:1061] Use fast generic engine


i:38


[WARNING 24-12-16 08:32:18.9352 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:18.9352 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:18.9352 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:19.6887 UTC kernel.cc:1233] Loading model from path /tmp/tmp2lv9rv25/model/ with prefix c5282bc25be2400e
[INFO 24-12-16 08:32:19.7025 UTC kernel.cc:1061] Use fast generic engine


i:39


[WARNING 24-12-16 08:32:20.1650 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:20.1650 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:20.1650 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:20.9609 UTC kernel.cc:1233] Loading model from path /tmp/tmpxn4wepl0/model/ with prefix b0855c4a1bde4d5e
[INFO 24-12-16 08:32:20.9736 UTC kernel.cc:1061] Use fast generic engine


i:40


[WARNING 24-12-16 08:32:21.4218 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:21.4218 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:21.4218 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:21.9302 UTC kernel.cc:1233] Loading model from path /tmp/tmpcmborry4/model/ with prefix 2940440768aa478c
[INFO 24-12-16 08:32:21.9344 UTC kernel.cc:1061] Use fast generic engine


i:41


[WARNING 24-12-16 08:32:22.3734 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:22.3734 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:22.3734 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:23.1840 UTC kernel.cc:1233] Loading model from path /tmp/tmp9zfpqwl3/model/ with prefix 1701a997024146d6
[INFO 24-12-16 08:32:23.1987 UTC kernel.cc:1061] Use fast generic engine


i:42


[WARNING 24-12-16 08:32:23.6387 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:23.6388 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:23.6388 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:24.2810 UTC kernel.cc:1233] Loading model from path /tmp/tmpxe5nwio0/model/ with prefix eb62b9e491f74df6
[INFO 24-12-16 08:32:24.2895 UTC kernel.cc:1061] Use fast generic engine


i:43


[WARNING 24-12-16 08:32:24.7225 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:24.7225 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:24.7226 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:25.5665 UTC kernel.cc:1233] Loading model from path /tmp/tmp6c8wrlfi/model/ with prefix 75b8b66dbc584935
[INFO 24-12-16 08:32:25.5829 UTC kernel.cc:1061] Use fast generic engine


i:44


[WARNING 24-12-16 08:32:26.0213 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:26.0214 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:26.0214 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:26.6450 UTC kernel.cc:1233] Loading model from path /tmp/tmp7jdkprsm/model/ with prefix 72d09d6a53c34441
[INFO 24-12-16 08:32:26.6538 UTC kernel.cc:1061] Use fast generic engine


i:45


[WARNING 24-12-16 08:32:27.0871 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:27.0871 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:27.0871 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:27.5735 UTC kernel.cc:1233] Loading model from path /tmp/tmpw49fyrsy/model/ with prefix 5b1c79978f2e4815
[INFO 24-12-16 08:32:27.5768 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:32:27.5768 UTC kernel.cc:1061] Use fast generic engine


i:46


[WARNING 24-12-16 08:32:28.0049 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:28.0049 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:28.0049 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:28.7606 UTC kernel.cc:1233] Loading model from path /tmp/tmpcygwnikg/model/ with prefix fb343953e57a456f
[INFO 24-12-16 08:32:28.7742 UTC kernel.cc:1061] Use fast generic engine


i:47


[WARNING 24-12-16 08:32:29.2119 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:29.2119 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:29.2119 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:29.9317 UTC kernel.cc:1233] Loading model from path /tmp/tmpdm7rqwdz/model/ with prefix a5ac75931fe042f4
[INFO 24-12-16 08:32:29.9442 UTC kernel.cc:1061] Use fast generic engine


i:48


[WARNING 24-12-16 08:32:30.3808 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:30.3809 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:30.3809 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:30.9030 UTC kernel.cc:1233] Loading model from path /tmp/tmpd1p00glq/model/ with prefix 8cb63fc5130544af
[INFO 24-12-16 08:32:30.9073 UTC kernel.cc:1061] Use fast generic engine


i:49


[WARNING 24-12-16 08:32:31.3353 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:31.3354 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:31.3354 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:31.9084 UTC kernel.cc:1233] Loading model from path /tmp/tmpb1cw19p7/model/ with prefix aa276a9cd08e4104
[INFO 24-12-16 08:32:31.9146 UTC kernel.cc:1061] Use fast generic engine


i:50


[WARNING 24-12-16 08:32:32.4022 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:32.4022 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:32.4023 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:33.1367 UTC kernel.cc:1233] Loading model from path /tmp/tmp_z0bfg_8/model/ with prefix f6633c93dc174464
[INFO 24-12-16 08:32:33.1481 UTC kernel.cc:1061] Use fast generic engine


i:51


[WARNING 24-12-16 08:32:33.5888 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:33.5889 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:33.5889 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:34.3707 UTC kernel.cc:1233] Loading model from path /tmp/tmpnsdd37kg/model/ with prefix 10ee60c4072446d3
[INFO 24-12-16 08:32:34.3854 UTC kernel.cc:1061] Use fast generic engine


i:52


[WARNING 24-12-16 08:32:34.8254 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:34.8255 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:34.8255 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:35.4091 UTC kernel.cc:1233] Loading model from path /tmp/tmp_pfe5_8h/model/ with prefix 7d0c12d31e7a4e12
[INFO 24-12-16 08:32:35.4164 UTC kernel.cc:1061] Use fast generic engine


i:53


[WARNING 24-12-16 08:32:35.8564 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:35.8564 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:35.8564 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:36.4307 UTC kernel.cc:1233] Loading model from path /tmp/tmpdveb6yho/model/ with prefix 21fbcd75526f4260
[INFO 24-12-16 08:32:36.4373 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-16 08:32:36.4378 UTC kernel.cc:1061] Use fast generic engine


i:54


[WARNING 24-12-16 08:32:36.8676 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:36.8676 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:36.8676 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:37.3427 UTC kernel.cc:1233] Loading model from path /tmp/tmperqrwjsj/model/ with prefix b304ac9b35b24994
[INFO 24-12-16 08:32:37.3458 UTC kernel.cc:1061] Use fast generic engine


i:55


[WARNING 24-12-16 08:32:37.7776 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:37.7776 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:37.7777 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:38.5493 UTC kernel.cc:1233] Loading model from path /tmp/tmp14vb4mpv/model/ with prefix ebaa997c17884010
[INFO 24-12-16 08:32:38.5628 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:32:38.5629 UTC kernel.cc:1061] Use fast generic engine


i:56


[WARNING 24-12-16 08:32:38.9946 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:38.9946 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:38.9946 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:39.6574 UTC kernel.cc:1233] Loading model from path /tmp/tmpp67ekyix/model/ with prefix 7e84cc844f884990
[INFO 24-12-16 08:32:39.6673 UTC kernel.cc:1061] Use fast generic engine


i:57


[WARNING 24-12-16 08:32:40.0939 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:40.0939 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:40.0939 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:40.6125 UTC kernel.cc:1233] Loading model from path /tmp/tmp1kduwr85/model/ with prefix d33de548fdda4151
[INFO 24-12-16 08:32:40.6164 UTC kernel.cc:1061] Use fast generic engine


i:58


[WARNING 24-12-16 08:32:41.0414 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:41.0414 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:41.0414 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:41.5919 UTC kernel.cc:1233] Loading model from path /tmp/tmp6mll6rnw/model/ with prefix b740156173384dd0
[INFO 24-12-16 08:32:41.5980 UTC kernel.cc:1061] Use fast generic engine


i:59


[WARNING 24-12-16 08:32:42.0329 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:42.0329 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:42.0329 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:42.6855 UTC kernel.cc:1233] Loading model from path /tmp/tmpp906csj5/model/ with prefix 135f762024e74b8b
[INFO 24-12-16 08:32:42.6942 UTC kernel.cc:1061] Use fast generic engine


i:60


[WARNING 24-12-16 08:32:43.1277 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:43.1277 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:43.1277 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:43.7731 UTC kernel.cc:1233] Loading model from path /tmp/tmpd55vo_ri/model/ with prefix 1d49ff9aa3c84992
[INFO 24-12-16 08:32:43.7822 UTC kernel.cc:1061] Use fast generic engine


i:61


[WARNING 24-12-16 08:32:44.2213 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:44.2213 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:44.2213 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:44.7655 UTC kernel.cc:1233] Loading model from path /tmp/tmpuanxtbhd/model/ with prefix 5d51537187564bd7
[INFO 24-12-16 08:32:44.7704 UTC kernel.cc:1061] Use fast generic engine


i:62


[WARNING 24-12-16 08:32:45.2165 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:45.2166 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:45.2166 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:45.9370 UTC kernel.cc:1233] Loading model from path /tmp/tmp2kxigikv/model/ with prefix 35f62defc84c40da
[INFO 24-12-16 08:32:45.9485 UTC kernel.cc:1061] Use fast generic engine


i:63


[WARNING 24-12-16 08:32:46.3913 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:46.3914 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:46.3914 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:47.0212 UTC kernel.cc:1233] Loading model from path /tmp/tmpqmmtioqz/model/ with prefix 66d8322de2844c87
[INFO 24-12-16 08:32:47.0300 UTC kernel.cc:1061] Use fast generic engine


i:64


[WARNING 24-12-16 08:32:47.4653 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:47.4653 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:47.4653 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:48.7498 UTC kernel.cc:1233] Loading model from path /tmp/tmp6xg1v969/model/ with prefix a1d4aa4a3294461b
[INFO 24-12-16 08:32:48.7597 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:32:48.7598 UTC kernel.cc:1061] Use fast generic engine


i:65


[WARNING 24-12-16 08:32:49.2845 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:49.2847 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:49.2847 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:49.8510 UTC kernel.cc:1233] Loading model from path /tmp/tmpeq1yalbs/model/ with prefix 4622e0b411b54708
[INFO 24-12-16 08:32:49.8564 UTC kernel.cc:1061] Use fast generic engine


i:66


[WARNING 24-12-16 08:32:50.3298 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:50.3298 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:50.3299 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:50.9257 UTC kernel.cc:1233] Loading model from path /tmp/tmpsc5q1cbl/model/ with prefix ea6e73b091e34439
[INFO 24-12-16 08:32:50.9328 UTC kernel.cc:1061] Use fast generic engine


i:67


[WARNING 24-12-16 08:32:51.3925 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:51.3926 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:51.3926 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:52.2549 UTC kernel.cc:1233] Loading model from path /tmp/tmplr8f1asp/model/ with prefix 4f2a4dbdda8a4549
[INFO 24-12-16 08:32:52.2750 UTC kernel.cc:1061] Use fast generic engine


i:68


[WARNING 24-12-16 08:32:52.7460 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:52.7461 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:52.7461 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:53.4527 UTC kernel.cc:1233] Loading model from path /tmp/tmpt8ct2kne/model/ with prefix c634331b809e442a
[INFO 24-12-16 08:32:53.4651 UTC kernel.cc:1061] Use fast generic engine


i:69


[WARNING 24-12-16 08:32:53.9293 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:53.9293 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:53.9294 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:54.4969 UTC kernel.cc:1233] Loading model from path /tmp/tmpeurx3gvf/model/ with prefix 68497765cceb44df
[INFO 24-12-16 08:32:54.5037 UTC kernel.cc:1061] Use fast generic engine


i:70


[WARNING 24-12-16 08:32:54.9536 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:54.9536 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:54.9537 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:55.5647 UTC kernel.cc:1233] Loading model from path /tmp/tmpmma34r6q/model/ with prefix 8d23a36d6abe486f
[INFO 24-12-16 08:32:55.5733 UTC kernel.cc:1061] Use fast generic engine


i:71


[WARNING 24-12-16 08:32:56.0202 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:56.0202 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:56.0203 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:56.6032 UTC kernel.cc:1233] Loading model from path /tmp/tmpnhdho5oi/model/ with prefix 660c01d301264afd
[INFO 24-12-16 08:32:56.6104 UTC kernel.cc:1061] Use fast generic engine


i:72


[WARNING 24-12-16 08:32:57.0564 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:57.0565 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:57.0565 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:57.8456 UTC kernel.cc:1233] Loading model from path /tmp/tmpvt2wjz6d/model/ with prefix 54ed93f7a3ac46f1
[INFO 24-12-16 08:32:57.8606 UTC kernel.cc:1061] Use fast generic engine


i:73


[WARNING 24-12-16 08:32:58.3038 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:58.3038 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:58.3038 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:58.8696 UTC kernel.cc:1233] Loading model from path /tmp/tmp4h7mw3ig/model/ with prefix aae561d277794cf2
[INFO 24-12-16 08:32:58.8762 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:32:58.8762 UTC kernel.cc:1061] Use fast generic engine


i:74


[WARNING 24-12-16 08:32:59.3085 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:59.3085 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:32:59.3085 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:32:59.9959 UTC kernel.cc:1233] Loading model from path /tmp/tmpg7gt33h2/model/ with prefix ff7d9d2b893445a0
[INFO 24-12-16 08:33:00.0067 UTC kernel.cc:1061] Use fast generic engine


i:75


[WARNING 24-12-16 08:33:00.4420 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:00.4420 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:00.4420 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:01.0602 UTC kernel.cc:1233] Loading model from path /tmp/tmp9d910uvs/model/ with prefix b510e846f3f14a92
[INFO 24-12-16 08:33:01.0677 UTC kernel.cc:1061] Use fast generic engine


i:76


[WARNING 24-12-16 08:33:01.5020 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:01.5020 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:01.5020 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:02.0219 UTC kernel.cc:1233] Loading model from path /tmp/tmp5bl7zdmf/model/ with prefix 3c0464817e4140ec
[INFO 24-12-16 08:33:02.0257 UTC kernel.cc:1061] Use fast generic engine


i:77


[WARNING 24-12-16 08:33:02.4687 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:02.4687 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:02.4688 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:02.9895 UTC kernel.cc:1233] Loading model from path /tmp/tmp_eg1lhe0/model/ with prefix 304e8be6ee574db6
[INFO 24-12-16 08:33:02.9942 UTC kernel.cc:1061] Use fast generic engine


i:78


[WARNING 24-12-16 08:33:03.4480 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:03.4480 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:03.4481 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:04.0986 UTC kernel.cc:1233] Loading model from path /tmp/tmpz9yn8l05/model/ with prefix d5d670b2e31c4749
[INFO 24-12-16 08:33:04.1078 UTC kernel.cc:1061] Use fast generic engine


i:79


[WARNING 24-12-16 08:33:04.5864 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:04.5864 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:04.5864 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:05.1712 UTC kernel.cc:1233] Loading model from path /tmp/tmpz_r3g4n_/model/ with prefix 4e53c7a7ad1d4e5b
[INFO 24-12-16 08:33:05.1781 UTC kernel.cc:1061] Use fast generic engine


i:80


[WARNING 24-12-16 08:33:05.6107 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:05.6108 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:05.6108 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:06.2503 UTC kernel.cc:1233] Loading model from path /tmp/tmpy8c_3sux/model/ with prefix fbc9673c76c24e71
[INFO 24-12-16 08:33:06.2595 UTC kernel.cc:1061] Use fast generic engine


i:81


[WARNING 24-12-16 08:33:06.6927 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:06.6927 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:06.6927 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:07.3698 UTC kernel.cc:1233] Loading model from path /tmp/tmpnkesp96n/model/ with prefix feb227d191a1446b
[INFO 24-12-16 08:33:07.3797 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-16 08:33:07.3803 UTC kernel.cc:1061] Use fast generic engine


i:82


[WARNING 24-12-16 08:33:07.8160 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:07.8161 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:07.8161 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:08.4374 UTC kernel.cc:1233] Loading model from path /tmp/tmppkopj0pw/model/ with prefix 1e47bc4867b7459f
[INFO 24-12-16 08:33:08.4465 UTC kernel.cc:1061] Use fast generic engine


i:83


[WARNING 24-12-16 08:33:08.8809 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:08.8809 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:08.8809 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:09.4823 UTC kernel.cc:1233] Loading model from path /tmp/tmpgsmw1p82/model/ with prefix 22776a8a94594deb
[INFO 24-12-16 08:33:09.4904 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:33:09.4904 UTC kernel.cc:1061] Use fast generic engine


i:84


[WARNING 24-12-16 08:33:09.9327 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:09.9327 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:09.9328 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:10.7361 UTC kernel.cc:1233] Loading model from path /tmp/tmpgq7nb8gk/model/ with prefix 8f6b48f6c0d94039
[INFO 24-12-16 08:33:10.7508 UTC kernel.cc:1061] Use fast generic engine


i:85


[WARNING 24-12-16 08:33:11.1839 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:11.1839 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:11.1839 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:11.7349 UTC kernel.cc:1233] Loading model from path /tmp/tmprr6lsmwg/model/ with prefix 3376db9727bf41c3
[INFO 24-12-16 08:33:11.7408 UTC kernel.cc:1061] Use fast generic engine


i:86


[WARNING 24-12-16 08:33:12.1778 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:12.1779 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:12.1779 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:12.9880 UTC kernel.cc:1233] Loading model from path /tmp/tmp3c1m3pz1/model/ with prefix f001ab122f544b2a
[INFO 24-12-16 08:33:13.0030 UTC kernel.cc:1061] Use fast generic engine


i:87


[WARNING 24-12-16 08:33:13.4404 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:13.4404 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:13.4404 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:14.3224 UTC kernel.cc:1233] Loading model from path /tmp/tmpaadmhjmb/model/ with prefix abfe2ba3e23e4bd8
[INFO 24-12-16 08:33:14.3388 UTC kernel.cc:1061] Use fast generic engine


i:88


[WARNING 24-12-16 08:33:14.7861 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:14.7861 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:14.7861 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:15.4711 UTC kernel.cc:1233] Loading model from path /tmp/tmpn0y_6nnq/model/ with prefix 2b15e033d6ad493f
[INFO 24-12-16 08:33:15.4816 UTC kernel.cc:1061] Use fast generic engine


i:89


[WARNING 24-12-16 08:33:15.9326 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:15.9327 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:15.9327 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:16.4525 UTC kernel.cc:1233] Loading model from path /tmp/tmp4e9wg93e/model/ with prefix d0a0e6d09759487a
[INFO 24-12-16 08:33:16.4571 UTC kernel.cc:1061] Use fast generic engine


i:90


[WARNING 24-12-16 08:33:16.8893 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:16.8894 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:16.8894 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:17.5422 UTC kernel.cc:1233] Loading model from path /tmp/tmpagx9c17a/model/ with prefix db5aa0e5d19c4e8c
[INFO 24-12-16 08:33:17.5517 UTC kernel.cc:1061] Use fast generic engine


i:91


[WARNING 24-12-16 08:33:18.7070 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:18.7070 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:18.7071 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:19.3385 UTC kernel.cc:1233] Loading model from path /tmp/tmp9lotzwef/model/ with prefix 1ed6f65b4b274198
[INFO 24-12-16 08:33:19.3468 UTC kernel.cc:1061] Use fast generic engine


i:92


[WARNING 24-12-16 08:33:19.8449 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:19.8450 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:19.8450 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:20.7311 UTC kernel.cc:1233] Loading model from path /tmp/tmptkpze291/model/ with prefix ffce176fe04c45e3
[INFO 24-12-16 08:33:20.7514 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-16 08:33:20.7515 UTC kernel.cc:1061] Use fast generic engine


i:93


[WARNING 24-12-16 08:33:21.2286 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:21.2286 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:21.2287 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:21.8930 UTC kernel.cc:1233] Loading model from path /tmp/tmpnz20xw6c/model/ with prefix c5805519e61c4ab3
[INFO 24-12-16 08:33:21.9040 UTC kernel.cc:1061] Use fast generic engine


i:94


[WARNING 24-12-16 08:33:22.3714 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:22.3714 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:22.3715 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:22.9618 UTC kernel.cc:1233] Loading model from path /tmp/tmpxnyuf8ma/model/ with prefix 9274c4a0f6694550
[INFO 24-12-16 08:33:22.9687 UTC kernel.cc:1061] Use fast generic engine


i:95


[WARNING 24-12-16 08:33:23.4219 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:23.4219 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:23.4220 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:24.0575 UTC kernel.cc:1233] Loading model from path /tmp/tmpmhmxmrpu/model/ with prefix 20cb1f09587a407e
[INFO 24-12-16 08:33:24.0660 UTC kernel.cc:1061] Use fast generic engine


i:96


[WARNING 24-12-16 08:33:24.5259 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:24.5260 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:24.5260 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:25.2106 UTC kernel.cc:1233] Loading model from path /tmp/tmp5tln0jjk/model/ with prefix 869edfdbabb84823
[INFO 24-12-16 08:33:25.2205 UTC kernel.cc:1061] Use fast generic engine


i:97


[WARNING 24-12-16 08:33:25.6757 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:25.6757 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:25.6757 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:26.2263 UTC kernel.cc:1233] Loading model from path /tmp/tmp7f5x5xcw/model/ with prefix b0b5d83b93194c10
[INFO 24-12-16 08:33:26.2313 UTC kernel.cc:1061] Use fast generic engine


i:98


[WARNING 24-12-16 08:33:26.6715 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:26.6715 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:26.6716 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:27.2550 UTC kernel.cc:1233] Loading model from path /tmp/tmppep7vzb_/model/ with prefix 9b7d046dfddf4229
[INFO 24-12-16 08:33:27.2626 UTC kernel.cc:1061] Use fast generic engine


i:99


[WARNING 24-12-16 08:33:27.7130 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:27.7130 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-16 08:33:27.7130 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-16 08:33:28.4759 UTC kernel.cc:1233] Loading model from path /tmp/tmpkn0a5jap/model/ with prefix ed41284086324011
[INFO 24-12-16 08:33:28.4898 UTC kernel.cc:1061] Use fast generic engine


Submission exported to /kaggle/working/submission.csv


In [ ]:
# What is next

If you want to learn more about TensorFlow Decision Forests and its advanced features, you can follow the official documentation [here](https://www.tensorflow.org/decision_forests) 